File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
/*
1. 0번째 숫자는 무조건 answer list에 추가해준다.
2. 만약 현재 list의 값이 그 전의 값과 다르다면 answer list에 추가, 같다면 추가하지 않음
*/
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int[] answer = {};
        ArrayList<Integer> arrList = new ArrayList<Integer>();
        
        for(int i=0; i<arr.length; i++){
            
            if(i==0){
                arrList.add(arr[i]);
            }
            
            else if(arr[i] != arr[i-1]){
                arrList.add(arr[i]);
            }
            
        }
        //answer 리스트 크기 선언후 Integer ArrayList을 int 배열로 변환
        answer = new int[arrList.size()];
        for (int i = 0 ; i < arrList.size() ; i++) {
            answer[i] = arrList.get(i).intValue();
        }
        return answer;
    }
}

File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
'''
1. 0번째 숫자는 무조건 answer list에 추가해준다.
2. 만약 현재 list의 값이 그 전의 값과 다르다면 answer list에 추가, 같다면 추가하지 않음
'''
def solution(arr):
    answer = []
    
    for i in range(len(arr)): 
        
        if i == 0:
            answer.append(arr[i])
        #elif arr[i] == arr[i-1]:
        #    continue
        elif arr[i] != arr[i-1]:
            answer.append(arr[i])    
        
    return answer